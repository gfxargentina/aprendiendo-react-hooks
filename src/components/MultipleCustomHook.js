import React from 'react'
import { useCounter } from '../hooks/useCounter';
import { useFetch } from '../hooks/useFetch'

export const MultipleCustomHook = () => {

    const { counter, increment } = useCounter(1);

   const { loading, data} = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);
   //console.log(state)
   const { author, quote } = !!data && data[0];

    return (
        <div className="container">
            <h1>BreakingBad Quotes</h1>
            <hr/>
            {loading ? (
                <div className="alert alert-warning text-center" >
                    Loading
                </div>
                ) : (
                    <blockquote className="blockquote text-right" >
                        <p className="mb-0"> { quote } </p>
                        <footer className="blockquote-footer" > {author} </footer>
                    </blockquote>
                )
            }

            <button 
            className="btn btn-primary"
            onClick={ increment }
             >
                Siguiente Frase
            </button>


        </div>
    )
}
