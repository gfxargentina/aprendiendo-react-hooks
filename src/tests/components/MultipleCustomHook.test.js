import React from 'react';
import { shallow } from 'enzyme';
import { MultipleCustomHook } from '../../components/MultipleCustomHook';
import { useFetch } from '../../hooks/useFetch';
import { useCounter } from '../../hooks/useCounter';

jest.mock('../../hooks/useFetch');
jest.mock('../../hooks/useCounter');


describe('Pruebas en <MultipleCustomHook />', () => {

    beforeEach( () =>  {
 
        useCounter.mockReturnValue({
            counter: 10,
            increment: () => {}
        });
 
    });

    test('debe de mostrarse correctamente', () => {

        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        });

        const wrapper = shallow( <MultipleCustomHook /> );
        expect( wrapper ).toMatchSnapshot();
        
    })
    
    test('Debe de mostrar la informacion', () => {

        useFetch.mockReturnValue({
            data: [{
                author: 'Luis',
                quote: 'Aprendiendo Testing'
            }],
            loading: false,
            error: null
        });

        const wrapper = shallow( <MultipleCustomHook /> );
        //console.log( wrapper.html() )
        expect( wrapper.find('.alert').exists() ).toBe(false);
        expect( wrapper.find('.mb-0').text().trim() ).toBe('Aprendiendo Testing');
        expect( wrapper.find('footer').text().trim() ).toBe('Luis');

        
        
    })
    
    
})
