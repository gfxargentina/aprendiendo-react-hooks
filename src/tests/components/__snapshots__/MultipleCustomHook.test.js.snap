// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHook /> debe de mostrarse correctamente 1`] = `
<div
  className="container"
>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-warning text-center"
  >
    Loading
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente Frase
  </button>
</div>
`;
